packages/vite/src/generators/configuration/schema.d.ts

```diff
@@ -1,5 +1,5 @@
 export interface ViteConfigurationGeneratorSchema {
-  uiFramework: 'react' | 'none';
+  uiFramework: 'react' | 'vue' | 'none';
```

Comment:
Why do we need to add `vue` here? It doesn't seem to be any different than `none`. 

In fact, why is `react` here? I think we should do whatever changes are necessary in `@nx/react`. And for `vue`, which there seems to be no changes needed, we should do it in `@nx/vue`.

I think we learned from `jest` that it's a bad idea to have multiple different configs for different frameworks within a tool plugin.

We can change this PR to not add `vue` here and pass `none` instead. Then followup with a PR to deprecate the `uiFramework` here.

Reply:
yep, makes sense!

Reply:
I made most changes needed to remove react, but still left react there for other reasons, which will be explained and maybe removed in future PR.